dace/transformation/interstate/transient_reuse.py

```diff
@@ -0,0 +1,184 @@
+from dace import registry, sdfg as sd
+from dace.graph import nodes
+from dace.transformation import pattern_matching
+from dace.properties import make_properties
+from sympy import N
+import networkx as nx
+
+
+def _atomic_counter_generator():
+    ctr = 0
+    while True:
+        ctr += 1
+        yield ctr
+
+
+_atomic_count = _atomic_counter_generator()
```

Comment:
@targetsm Please don't use atomic counters or other design patterns that run on global code. When you add an array or data descriptor you have the `find_new_name` flag that you can set (so "transient_reuse" becomes "transient_reuse_1", 2, ...)